packages/common/base/src/apiClient/CrossmintApiClient.ts

```diff
@@ -0,0 +1,52 @@
+import { validateAPIKey } from "@/apiKey/validateAPIKey";
+import { ValidateAPIKeyPrefixExpectations, ValidateAPIKeyPrefixSuccessResult } from "@/apiKey/validateAPIKeyPrefix";
+
+import { ApiClient } from "./ApiClient";
+
+export type CrossmintApiClientCtorParams = CrossmintApiClientCtorWithoutExpectationsParams & {
+    expectations?: ValidateAPIKeyPrefixExpectations;
+};
+
+export type CrossmintApiClientCtorWithoutExpectationsParams = {
+    apiKey: string;
+    overrideBaseUrl?: string;
+};
+
+export class CrossmintApiClient extends ApiClient {
+    protected parsedAPIKey: ValidateAPIKeyPrefixSuccessResult;
+    protected overrideBaseUrl?: string;
+
+    constructor({ apiKey, expectations, overrideBaseUrl }: CrossmintApiClientCtorParams) {
+        const apiKeyValidationResult = validateAPIKey(apiKey, expectations);
+        if (!apiKeyValidationResult.isValid) {
+            throw new Error(apiKeyValidationResult.message);
+        }
+        super({ "x-api-key": apiKey });
+        this.parsedAPIKey = apiKeyValidationResult;
+        this.overrideBaseUrl = overrideBaseUrl;
+    }
+
+    get baseUrl() {
+        let url: string;
+        if (this.overrideBaseUrl) {
+            url = this.overrideBaseUrl;
+        }
+
+        const { environment } = this.parsedAPIKey;
+        switch (environment) {
+            case "production":
+                url = "https://www.crossmint.com";
+                break;
+            case "staging":
+                url = "https://staging.crossmint.com";
+                break;
+            case "development":
+                url = "http://localhost:3000";
+                break;
```

Comment:
I would add a comment here making it explicit this is a utility for us crossmint devs